### ISSUE TITLE
Lacks project-aware context

### ISSUE THREAD

**Issue by nirzaf** [microsoft/vscode-copilot-release#237]
Very bad in understanding the context of the code - GitHub Copilot doing better job than this

**Comment by RogerBarreto** [comment]
@ulugbekna I think your question already answers it.

Basically would be important when using Chat to be able to specify in the interface without having to text, what is the context. 

Open (current one) / Workspace Project specific / Current open file.

**Comment by nirzaf** [comment]
@ulugbekna Not understanding the libraries and NuGet packages we are using, sample codes popping with totally irrelevant NuGet packages

**Comment by neilime** [comment]
I've tried to given direct absolute path to help copilot to understand the context but no effect at all:

By exemple:

```
In the current workspace project (/home/my/project): Can you implement the missing method "method1" in the file "/home/my/project/MyClass.php".

Please refer to "/home/my/project/MyReferenceClass.php" to match with expected behavior.
```

But copilot generated basic non contextualized code without any relevant code, in terms of project context or even regarding the ReferenceClass

**Comment by fisforfaheem** [comment]
Yes it gives errors sometimes while chatting as well and cant understand longer boilerplate code so it should be improved

**Comment by ulugbekna** [comment]
duplicate of #14

**Comment by digitarald** [comment]
Sharing from #14: `@workspace` with _"Used references"_ landed in Copilot Chat 0.10 as part of VS Code's [October release](https://code.visualstudio.com/updates/v1_84#_workspace)

<img width="657" alt="image" src="https://github.com/microsoft/vscode-copilot-release/assets/8599/9ac45254-74ae-4319-8c57-a612221b6985">

Please give it a try and file new issues for any follow-up feedback!
